src/form-builder/AddressField.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {useEffect} from 'react';
+import { useField, FieldHookConfig, useFormikContext } from 'formik';
+
+import { chainValidations, required } from '../utils/validation';
+import { AddressProps, Address } from './types';
+import {
+    TextField,
+    CheckboxField,
+    SelectField
+} from './index'
+import {
+    Countries,
+    CountryUSA,
+    States,
+    MilitaryCities,
+    MilitaryStates
+ } from '../utils/constants'
+
+const AddressField = (props: AddressProps): JSX.Element => {
+
+    const withValidation = {
+      ...props,
+      validate: chainValidations(props, [required]),
+    };
+    const [field] = useField(withValidation as FieldHookConfig<string>);
+    const {setFieldValue} = useFormikContext();
+
+    if(!field.value) {
```

Comment:
This condition was added because testing was failing with value being undefined.